src/signature/parser.sml

```diff
@@ -0,0 +1,99 @@
+(* The grammar can be found at
+   https://github.com/JonPRL/sml-red-jonprl/blob/master/doc/signatures.pdf *)
+
+structure SignatureParser =
+struct
+
+  open ParserCombinators
+  open CharParser
+
+  open StringSignature
+
+  infixr 4 << >>
+  infixr 3 &&
+  infix  2 -- ##
+  infix  2 wth suchthat return guard when
+  infixr 1 || <|> ??
+
+  fun inBrackets p = char #"[" >> p << char #"]"
+  fun inParentheses p = char #"(" >> p << char #")"
+  fun inBraces p = char #"{" >> p << char #"}"
+
+  val skipChars = repeatSkip (space <|> char #"\t" <|> char #"\n")
+
+  val opid = repeat1 letter wth String.implode
+  val sortid = repeat1 letter wth String.implode
+  val term = repeat1 letter wth String.implode
```

Comment:
Probably we can't use `letter` here, since we want to parse arbitrary strings of any characters in (later on, to be parsed into actual terms in a separate phase of parsing/elaboration).

Can we use a parser that always succeeds here?